luaPlugin/luaP.cpp

```diff
@@ -1379,56 +1590,185 @@ sol::state* luaP::init(std::string& luaFilePath, std::string& modPath)
+	@tfield buildingInQueue[6] buildingInQueue index+1 (1-6), see usage for getQueueName()
+
+	@table buildingsQueue
+	*/
+	types.buildingsQueue = luaState.new_usertype<buildingsQueue>("buildingsQueue");
+	types.buildingsQueue.set("firstIndex", &buildingsQueue::firstIndex);
+	types.buildingsQueue.set("lastIndex", &buildingsQueue::lastIndex);
+	types.buildingsQueue.set("currentBuildingPosition", &buildingsQueue::currentBuildingIndex);
+	types.buildingsQueue.set("numBuildingsInQueue", &buildingsQueue::buildingsInQueue);
+	types.buildingsQueue.set("buildingInQueue", sol::property([](buildingsQueue& self) { return std::ref(self.items); }));
+
+
+	///BuildingInQueue table section
+	//@section buildingInQueueTable
+
+	/***
+	Basic buildingInQueue table
+
+	@tfield building building nil if building doesn't exist yet
+	@tfield settlementStruct settlement
+	@tfield int currentLevel
+	@tfield int previousLevel
+	@tfield int buildCost
+	@tfield int buildTurnsPassed
+	@tfield int buildTurnsRemaining
+	@tfield int percentBuilt
+	@tfield getQueueType getQueueType
+	@tfield getQueueName getQueueName
+
+	@table buildingInQueue
+	*/
+	types.buildingInQueue = luaState.new_usertype<buildingInQueue>("buildingInQueue");
+	types.buildingInQueue.set("buildingData", &buildingInQueue::buildingData);
+	types.buildingInQueue.set("building", &buildingInQueue::existsBuilding);
+	types.buildingInQueue.set("settlement", &buildingInQueue::settlement);
+	types.buildingInQueue.set("currentLevel", &buildingInQueue::currentLevel);
+	types.buildingInQueue.set("previousLevel", &buildingInQueue::pastLevel);
+	types.buildingInQueue.set("buildCost", &buildingInQueue::buildCost);
+	types.buildingInQueue.set("buildTurnsPassed", &buildingInQueue::buildTurnsPassed);
+	types.buildingInQueue.set("buildTurnsRemaining", &buildingInQueue::turnsToBuild);
+	types.buildingInQueue.set("percentBuilt", &buildingInQueue::petcentBuilded);
 	/***
 	Get settlement religion value.
 	It is desirable that after your intervention, the sum of the influence of all religions does not exceed 1.
 	@function settlementStruct:getReligion
 	@tparam int religionID in order of descr_religions.txt, started from 0
 	@treturn float religionValue from 0 to 1
 	@usage
-	local firstRelVal=settlementStruct:getReligion(0);//get value of religion with ID 0
+	--see usage for getQueueName()
 	*/
-	types.settlementStruct.set_function("getReligion", &settlementHelpers::getReligion);
+	types.buildingInQueue.set_function("getQueueType", &buildingStructHelpers::getQueueType);
 	/***
 	Set settlement religion value.
 	It is desirable that after your intervention, the sum of the influence of all religions does not exceed 1.
 	@function settlementStruct:setReligion
 	@tparam int religionID in order of descr_religions.txt, started from 0
 	@tparam float religionValue from 0 to 1
 	@usage
-	settlementStruct:setReligion(0,0.5);//set religion with ID 0 as 50%
+	function printBuildQueueInfo(i, j, sett, resultList)
```

Comment:
Good suggestion. I will try to write a helper function so that the user can do: <ourQueue:getBuildingInQueue[i]> where i = n is the building in the nth _position_ regardless of index.